charts/alfresco-common/templates/_helpers-url.tpl

```diff
@@ -0,0 +1,45 @@
+{{/*
+Build up CSRF referer
+*/}}
+{{- define "alfresco-common.csrf.referer" -}}
+{{- $known_urls := . }}
+{{- $csrf_referers := list }}
+{{- if kindIs "string" . }}
+  {{- $known_urls = splitList "," . }}
+{{- end }}
+{{- range $known_urls }}
+  {{- $parsed_url := urlParse . }}
+  {{- $known_url := urlJoin (dict "host" $parsed_url.host "path" $parsed_url.path "scheme" $parsed_url.scheme) }}
+  {{- $csrf_referers = append $csrf_referers $known_url }}
+{{- end }}
+{{- $csrf_referers | join "/.*\\|" }}/.*
+{{- end -}}
+
+{{/*
+Build up CSRF Origin
+*/}}
+{{- define "alfresco-common.csrf.origin" -}}
+{{- $known_urls := . }}
+{{- $csrf_origins := list }}
+{{- if kindIs "string" . }}
+  {{- $known_urls = splitList "," . }}
+{{- end }}
+{{- range $known_urls }}
+  {{- $parsed_url := urlParse . }}
+  {{- $known_url := urlJoin (dict "host" $parsed_url.host "path" $parsed_url.path "scheme" $parsed_url.scheme) }}
+  {{- $csrf_origins = append $csrf_origins $known_url }}
+{{- end }}
+{{- $csrf_origins | join "," }}
+{{- end -}}
```

Comment:
isn't most of the code in common with `alfresco-common.csrf.referer`?

Reply:
actually my definition of origin was wrong. The only part that's common is the url split which is not very interesting to turn into a named template of its own as it involves turning output into json and reading from json where it needs to be used (given template only output strings)

Reply:
there is a chance to have some unit tests here? those functions are hard to understand what they do otherwise

Reply:
not using `helm unittest` as it's a library chart. The consumer charts should implement these tests maybe?

Reply:
Some unit tests in the share chart: https://github.com/Alfresco/alfresco-helm-charts/blob/6b4bdc011c41ad6069e3b64b4673d0ad036b4583/charts/alfresco-share/tests/cm_test.yaml#L15